src/hopen/syntax/silly_j.cljc

```diff
@@ -0,0 +1,54 @@
+(ns hopen.syntax.silly-j
+  (:require [hopen.renderer.xf :as rxf]
+            [instaparse.core :as insta]))
+
+(declare parse)
+
+;; TODO test case
+(defn render
+  ([template data]
+   (render template data '()))
+  ([template data other-env]
+   (let [new-env (update rxf/default-env :bindings assoc other-env)
+         renderer (rxf/renderer (parse template) new-env)]
+     (transduce renderer str data))))
+
+(def ebnf
+  (insta/parser "
+                S       = [STR | SILLYSTR]+
+                STR      = #'[^{}]+'
+                SILLYSTR =  OBK SILLY CBK
+                OBK      = '{'
+                CBK      = '}'
+                SILLY    = [FCTX | FN]
+                FCTX     = '@' ':' ATTR
+                ATTR     = #'[^}]'+
+                FN       = FNNAME APPLIES*
+                APPLIES  = SPC | FCTX | APPLY
+                FNNAME   = #'[a-zA-Z0-9\\-]+'
+                SPC      = #'\\ '+
+                APPLY    = #'^[^@\\ ][^}\\ ]+'
+                "
+                ))
```

Comment:
@Odie I think that you might want to give instaparse a try.

Reply:
Well, I think mustache is sufficiently simple to deal with by hand. (Famous last words?)

Instaparse is really fantastic, but my last impression of it was “really magical, but maybe somewhat slow”. It might have been due to an ambiguous grammar that caused it to try going down two separate paths of the grammar tree though.

Anyway, since there are no complicated grammatical structures, and mustache is already fully spec’ed out, I’ll try seeing it through the rest of the way.